### PR TITLE
feat: updated cli config accroding to portal config

### LINKED ISSUE #1076: Update Redocly config validation
Redocly CLI should validate the universal Redocly config correctly. We have to update the type tree to be in sync with JSONSchema we use to validate the config in Portal.

### PR DESCRIPTION
**Pull request by Skavengerr**
## What/Why/How?
Fixes Issue #1076 by updating the Redocly CLI configuration, which was copied from the portal config.

## Reference

Fixes #1076

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/types/redocly-yaml.ts [hunk 1]
```diff
@@ -151,12 +152,338 @@ const RootConfigStyleguide: NodeType = {
...
+    off: { type: 'boolean', default: false },
+    position: { type: 'string', enum: ['first', 'last', 'replace', 'off'], default: 'first' },
+    strictExamples: { type: 'boolean', default: false },
+    errorIfForcedExampleNotFound: { type: 'boolean', default: false },
+    description: { type: 'string' },
+  },
+};
+
+
+export const ConfigRoot = {
```
Comment (tatomyr) [inline packages/core/src/types/redocly-yaml.ts:482]: Why removed the type?

```suggestion
export const ConfigRoot: NodeType = {
```

File: packages/core/src/types/redocly-yaml.ts [hunk 2]
```diff
@@ -151,12 +152,338 @@ const RootConfigStyleguide: NodeType = {
...
   properties: {
     organization: { type: 'string' },
-    apis: 'ConfigApis',
     ...RootConfigStyleguide.properties,
-    theme: 'ConfigRootTheme',
+    theme: { type: 'object', default: {} },
```
Comment (tatomyr) [inline packages/core/src/types/redocly-yaml.ts:486]: I'd prefer extending 'ConfigRootTheme' type if it's possible. BTW, do we still use it somewhere?

File: packages/core/src/types/redocly-yaml.ts [hunk 3]
```diff
@@ -172,6 +499,17 @@ const ConfigRoot: NodeType = {
         type: 'string',
       },
     },
+    redirects: { type: 'object', additionalProperties: redirectConfigSchema, default: {} },
+    licenseKey: { type: 'string' },
+    apis: apiConfigSchema,
```
Comment (tatomyr) [inline packages/core/src/types/redocly-yaml.ts:504]: Why not extending the 'ConfigApis' type?

File: packages/core/src/types/redocly-yaml.ts [hunk 4]
```diff
@@ -54,7 +55,7 @@ const builtInRulesList = [
   'scalar-property-missing-example',
   'spec-components-invalid-map-name',
   'required-string-property-missing-min-length',
-  'spec-strict-refs',
+  'spec-ref-validation',
```
Comment (tatomyr) [inline packages/core/src/types/redocly-yaml.ts:58]: We renamed this in the previous release. Why it's reverted back?

File: packages/core/src/types/redocly-yaml.ts [hunk 5]
```diff
@@ -141,7 +142,7 @@ const ConfigStyleguide: NodeType = {
   },
 };
 
-const RootConfigStyleguide: NodeType = {
+export const RootConfigStyleguide: any = {
```
Comment (tatomyr) [inline packages/core/src/types/redocly-yaml.ts:145]: Could you avoid  the `any` type?

**Review by tatomyr — CHANGES_REQUESTED** [review]

File: packages/core/src/types/redocly-yaml.ts (hunk 4 above)
Comment (tatomyr) [reply packages/core/src/types/redocly-yaml.ts:62]: I still see the wrong name. It should be `'spec-strict-refs'` instead.

File: __tests__/lint/spec-strict-refs/snapshot.js [hunk 6]
```diff
@@ -1,46 +1,21 @@
...
-18 | type: object
-19 | properties:
-20 |   $ref: './props.yaml'
-   |   ^^^^
-21 |   name:
-22 |     type: string
-
-Error was generated by the spec-strict-refs rule.
-
+exports[`E2E lint spec-strict-refs 1`] = `
```
Comment (tatomyr) [inline __tests__/lint/spec-strict-refs/snapshot.js:3]: Why this has been changed?

**Review by tatomyr — CHANGES_REQUESTED** [review]

File: __tests__/lint/spec-strict-refs/snapshot.js (hunk 6 above)
Comment (IgorKarpiuk) [reply __tests__/lint/spec-strict-refs/snapshot.js:3]: @tatomyr `E2E lint spec-ref-validation 1` from old snapshot. and this one  just changed location from [here](https://github.com/Redocly/redocly-cli/pull/1133/files?diff=split&w=0#diff-3fdf21f2551fc727676e37674367dfa62ed3434527db5aecfe04a9daf270a696L42)

File: packages/core/src/types/redocly-yaml.ts [hunk 7]
```diff
@@ -141,40 +142,6 @@ const ConfigStyleguide: NodeType = {
   },
 };
 
-const RootConfigStyleguide: NodeType = {
```
Comment (tatomyr) [inline packages/core/src/types/redocly-yaml.ts:144]: Can we get this and `ConfigRoot` back in **redocly-yaml.ts** and move [this line](https://github.com/Redocly/redocly-cli/pull/1133/files?diff=split&w=1#diff-2528117ea3cbe25dfbfd0583c9b11ba7baa7ef347184dfba32a73ec486aa1a5eR186) to the **config-root.ts** file instead?

File: packages/core/src/types/config-root.ts [hunk 8]
```diff
@@ -0,0 +1,376 @@
+import { NodeType } from '.';
```
Comment (tatomyr) [inline packages/core/src/types/config-root.ts:1]: I suggest renaming this file into **external-config-schemas.ts** or something like that.

**Review by tatomyr — COMMENTED** [review]
Please keep the existing **redocly-yaml.ts** with the minimal changes and move the extra config schemas in the separate file.

**Pull request merged**